pkg/aws/ec2/ec2.go

```diff
@@ -309,10 +312,16 @@ func (c *Client) CreateNetworkInterface(toAllocate int64, subnetID, desc string,
 	resp, err := create.Send()
 	c.metricsAPI.ObserveEC2APICall("CreateNetworkInterfaceRequest", deriveStatus(create.Request, err), sinceStart.Seconds())
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
-	return *resp.NetworkInterface.NetworkInterfaceId, nil
+	_, eni, err := parseENI(resp.NetworkInterface, nil, nil)
+	if err != nil {
+		return *resp.NetworkInterface.NetworkInterfaceId, nil, nil
```

Comment:
this does not seem quite right... why are we ignoring the error here?

Reply:
Because returning the ENI information is optional. If the allocation succeeded but the parsing failed, we shouldn't revert the allocation but wait for the EC2 sync to pull in the information.

Reply:
Please add a comment stating that, it's not obvious.